src/get-authentication-token.ts

```diff
@@ -1,19 +1,33 @@
+import * as github from "@actions/github";
+import { createAppAuth } from "@octokit/auth-app";
 import { Octokit } from "@octokit/rest";
-import { GITHUB_TOKEN } from "./configuration/constants";
+import { UBIQUIBOT_APP_ID, INSTALLATION_ID, UBIQUIBOT_APP_PRIVATE_KEY } from "./configuration/constants";
 
 let octokitInstance: Octokit | null = null;
 
-function getAuthenticationToken(): string {
-  const auth = GITHUB_TOKEN;
-  if (!auth) {
-    throw new Error("No authentication token provided");
+async function getAuthenticationToken() {
+  const appId = UBIQUIBOT_APP_ID;
+  const privateKey = UBIQUIBOT_APP_PRIVATE_KEY;
+  // Defaults to the ENV value in case we are not in a GitHub Action context
+  const inputs = {
+    installationId: INSTALLATION_ID,
+  };
+
+  if (github.context.payload.inputs) {
+    const eventPayload = JSON.parse(github.context.payload.inputs.eventPayload);
+    if (eventPayload?.installation) {
+      inputs.installationId = eventPayload.installation.id;
+    }
   }
-  return auth;
+
+  const auth = createAppAuth({ appId, privateKey, installationId: inputs.installationId });
```

Comment:
app id and private key won't be provided to plugins but instead a JWT token will be passed via inputs